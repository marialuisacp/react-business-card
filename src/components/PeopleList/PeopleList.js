import React, { Component } from 'react';
import PropTypes from 'prop-types';
import CardPerson from '../PersonCard/PersonCard';
import { withRouter } from "react-router-dom";

import './PeopleList.scss';
import '../../assets/styles/styles.scss';

class PeopleList extends Component {
  constructor(props) {
    super(props);
    this.goToDetails = this.goToDetails.bind(this);
  }

  goToDetails = (data) => {
    this.props.history.push({
      pathname: '/detail',
      state: { data: data }
    });
  }

  render() {
    const { data } = this.props;
    const outlineCards = [];

    const outlineCardsLength = 5 - (data.length);
    for (let i = 0; i < outlineCardsLength; i++) {
      outlineCards.push({});
    };
    return (
      <div className='list-professionals' id='list-people-component'>
        {(data && data.length
          && data.map((person, k) => (<div onClick={() => this.goToDetails(person)} key={k}><CardPerson person={person} /></div>))) || ''}
        {(outlineCards && outlineCards.length
          && outlineCards.map((person, k) => (<CardPerson key={k} person={person} />))) || ''}
      </div >
    );
  }
}

PeopleList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object)
};

export default withRouter(PeopleList);