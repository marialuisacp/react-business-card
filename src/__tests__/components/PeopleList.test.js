import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import PeopleList from '../../components/PeopleList/PeopleList';

describe('PeopleList', () => {
  test('Component PeopleList should render correctly', () => {
    const props = {
      data: [
        {
          letter: 'D',
          people: [{
            name: 'Danilo Rocha',
            office: 'Desenvolvedor',
            image: './assets/images/people/d01.jpg',
            email: 'danilo.rocha@mundiale.com'
          }]
        },
        {
          letter: 'E',
          people: [{
            name: 'Eduardo Viana',
            office: 'UX Designer',
            image: './assets/images/people/e01.jpg',
            email: 'eduardo.viana@mundiale.com'
          }]
        }]
    };

    const component = renderer.create(
      <Router>
        <PeopleList {...props} />
      </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
});
