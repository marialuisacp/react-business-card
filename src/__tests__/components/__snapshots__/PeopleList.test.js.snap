// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleList Component PeopleList should render correctly 1`] = `
<div
  className="list-professionals list-people-component"
>
  <div
    onClick={[Function]}
  >
    <div
      className="card-person card-info"
    >
      <div
        className="circle"
      />
      <div
        className="card outline"
      />
    </div>
  </div>
  <div
    onClick={[Function]}
  >
    <div
      className="card-person card-info"
    >
      <div
        className="circle"
      />
      <div
        className="card outline"
      />
    </div>
  </div>
  <div
    className="card-person card-info"
  >
    <div
      className="circle"
    />
    <div
      className="card outline"
    />
  </div>
  <div
    className="card-person card-info"
  >
    <div
      className="circle"
    />
    <div
      className="card outline"
    />
  </div>
  <div
    className="card-person card-info"
  >
    <div
      className="circle"
    />
    <div
      className="card outline"
    />
  </div>
</div>
`;
